provider/provider_test.go

```diff
@@ -44,8 +45,25 @@ func TestConfigure(t *testing.T) {
 	})
 	require.NoError(t, err)
 
-	_, err = p.Configure(ctx, &pulumirpc.ConfigureRequest{
-		Args: args,
+	// Ideally we would just call p.Configure directly, but we need the
+	// integration server to inject runtime info for us.
+	argsMap, err := plugin.UnmarshalProperties(args, plugin.MarshalOptions{})
+	require.NoError(t, err)
+	s := integration.NewServer("docker-build", semver.Version{Major: 0}, provider.Provider{
+		// Roundabout way to get the integration server to invoke our outermost
+		// Configure RPC endpoint.
+		Configure: func(ctx context.Context, req provider.ConfigureRequest) error {
+			args, err := plugin.MarshalProperties(req.Args, plugin.MarshalOptions{})
+			require.NoError(t, err)
+			_, err = p.Configure(ctx, &rpc.ConfigureRequest{
+				Variables: req.Variables,
+				Args:      args,
+			})
+			return err
+		},
+	})
+	err = s.Configure(provider.ConfigureRequest{
+		Args: argsMap,
```

Comment:
`Configure` as currently set up uses `RunInfo` to improve error messages. I'm not sure why you create a server and then run configure inside it. Why not do this?
```suggestion
	args, err := plugin.MarshalProperties(req.Args, plugin.MarshalOptions{})
	require.NoError(t, err)
	s := integration.NewServer("docker-build", semver.Version{Major: 0}, p)
	err = s.Configure(provider.ConfigureRequest{
		Args: args,
	})
```

Reply:
The provider in this case is a `rpc.ResourceProviderServer` in order to override `Configure` and deserialize the old JSON config before it hits go-provider.